### PR TITLE
Remove non-Favorites lists

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-819

## What changed (plus any additional context for devs)
In this PR: 
Remove UI + logic for non-favorites lists (trending, watchlist, defi, stablecoins)

In the followup PRs:
Minor UI updates, refactor of remaining logic

Specifics:
1. Removed (and deleted) `Lists` component from `DiscoverHome`
2. Removed non-favorites lists from `AddTokenSheet`
3. Transitioned all access/modification to favorites from `userLists` redux to `uniswap` redux. The favorites list's existence/handling in `userLists` redux is just a wrapper around `uniswap` redux.
4. Deleted all `userLists` redux
5. Deleted `useUserLists` hook
6. Commented out obsolete migration involving userLists (pretty sure the empty migration itself must stay so it doesn't mess up migration tracking state)
7. Removed obsolete i18n
8. Removed obsolete types

## Screen recordings / screenshots

https://github.com/rainbow-me/rainbow/assets/15272675/6089a629-fab4-4302-bd16-bd8e3d0143e3

## What to test
* adding/removing favorites works
* there's no trace of other lists (if there is, app will definitely break)

### CODE REVIEW

**Comment by benisgold** [comment]
don't worry, the redux state usage that was added will be removed shortly

**Comment by jinchung** [comment]
Can you also remove any relevant test for lists in e2e/discoverSheetFlow.spec.js

File: src/components/exchange/ExchangeAssetList.tsx [hunk 1]
```diff
@@ -305,7 +303,7 @@ const ExchangeAssetList: ForwardRefRenderFunction<
             setLocalFavorite(prev => {
               const address = rowData.address;
               const newValue = !prev?.[address];
-              updateList(address, 'favorites', newValue);
+              dispatch(uniswapUpdateFavorites(address, newValue));
```
Comment (skylarbarrera) [inline src/components/exchange/ExchangeAssetList.tsx:306]: this is gonna have conflicts post other PR yeah?

**Review by skylarbarrera — APPROVED** [review]
👑

**Review by ibrahimtaveras00 — APPROVED** [review]
looks good 👍🏽

**Pull request merged**